src/components/EnterpriseList/index.jsx

```diff
@@ -3,178 +3,96 @@ import PropTypes from 'prop-types';
 
 import H1 from '../H1';
-import StatusAlert from '../StatusAlert';
+import TableContainer from '../../containers/TableContainer';
+import LmsApiService from '../../data/services/LmsApiService';
 import SearchBar from '../SearchBar';
-import LoadingMessage from '../LoadingMessage';
-import TableWithPagination from '../TableWithPagination';
-
-import { formatTableOptions } from '../../utils';
 
 class EnterpriseList extends React.Component {
   constructor(props) {
     super(props);
 
-    const { enterprises, location } = props;
-
-    const queryParams = this.formatQueryParams(qs.parse(location.search));
-
+    const { location } = props;
+    const queryParams = qs.parse(location.search);
     this.state = {
-      columns: [
-        {
-          label: 'Enterprise',
-          key: 'link',
-        },
-      ],
-      enterprises: enterprises && enterprises.results,
-      pageCount: enterprises && enterprises.num_pages,
       searchQuery: queryParams.search || '',
       searchSubmitted: !!queryParams.search,
     };
-
-    this.formatEnterpriseData = this.formatEnterpriseData.bind(this);
   }
 
   componentDidMount() {
-    const options = qs.parse(this.props.location.search);
-    this.clearPortalConfiguration();
-    this.getEnterpriseList(formatTableOptions(options));
-    this.props.getLocalUser();
-  }
-
-  componentDidUpdate(prevProps) {
-    const { enterprises, searchQuery } = this.props;
-
-    if (enterprises !== prevProps.enterprises) {
-      this.setState({ // eslint-disable-line react/no-did-update-set-state
-        enterprises: enterprises && enterprises.results,
-        pageCount: enterprises && enterprises.num_pages,
-      });
-    }
-    if (searchQuery !== prevProps.searchQuery) {
-      const options = this.formatQueryParams({ search: searchQuery });
-      this.props.history.push(`?${qs.stringify(options)}`);
-      this.getEnterpriseList(options);
-    }
-  }
-
-  getEnterpriseList(options) {
-    this.props.getEnterpriseList(options);
-  }
-
-  clearPortalConfiguration() {
     this.props.clearPortalConfiguration();
-  }
-
-  formatQueryParams(options) {
-    return omitBy({
-      search: options.search,
-    }, isEmpty);
-  }
-
-  formatEnterpriseData(enterprises) {
-    if (!enterprises) {
-      return null;
-    } else if (!enterprises.length) {
-      return [];
-    }
-
-    return enterprises.map(enterprise => ({
-      link: <Link to={`/${enterprise.slug}/admin`}>{enterprise.name}</Link>,
-      name: enterprise.name,
-      slug: enterprise.slug,
-      uuid: enterprise.uuid,
-    }));
+    this.props.getLocalUser();
   }
 
   handleSearch(query) {
-    this.props.setSearchQuery(query);
     this.setState({
       searchQuery: query,
       searchSubmitted: true,
     });
+    this.props.searchEnterpriseList({
+      // Ternary used to turn empty string ('') into undefined so it does not get set in qs
+      search: query ? query : undefined, // eslint-disable-line no-unneeded-ternary
```

Comment:
Could this actually be `search: query || undefined,` to avoid the eslint disable?